actionview/lib/action_view/helpers/tags/base.rb

```diff
@@ -79,35 +79,22 @@ def add_default_name_and_id_for_value(tag_value, options)
         end
 
         def add_default_name_and_id(options)
-          if options.has_key?("index")
-            options["name"] ||= options.fetch("name"){ tag_name_with_index(options["index"], options["multiple"]) }
-            options["id"] = options.fetch("id"){ tag_id_with_index(options["index"]) }
-            options.delete("index")
-          elsif defined?(@auto_index)
-            options["name"] ||= options.fetch("name"){ tag_name_with_index(@auto_index, options["multiple"]) }
-            options["id"] = options.fetch("id"){ tag_id_with_index(@auto_index) }
-          else
-            options["name"] ||= options.fetch("name"){ tag_name(options["multiple"]) }
-            options["id"] = options.fetch("id"){ tag_id }
-          end
-
-          options["id"] = [options.delete('namespace'), options["id"]].compact.join("_").presence
-        end
-
-        def tag_name(multiple = false)
-          "#{@object_name}[#{sanitized_method_name}]#{"[]" if multiple}"
+          index = name_and_id_index(options)
+          options["name"] ||= options.fetch("name"){ tag_name(options["multiple"], index) }
+          options["id"] = options.fetch("id"){ tag_id(index, options.delete("namespace")) }
```

Comment:
Reworked so that behavior left unchanged.

But it is currently very tricky. For example if `{namespace: 'something', id: nil}` it results in HTML id being equal `something`, but I would expect `nil` instead, because  it will lead to HTML id collision, that namespace is designed to avoid.

I didn't change that but it looks wrong.